### PR TITLE
Cleanups and robustness improvements in stats processing

### PR DESCRIPTION
**Pull request by mpetazzoni**
Cleanup code to not use @classmethod anymore. Be more robust in
processing memory stats if detailed memory statistics are somehow not
returned by Docker.

### CODE REVIEW

**Comment by mpetazzoni** [comment]
@mdubbyap Feel free to take a look too.

File: dockerplugin.py [hunk 1]
```diff
@@ -51,109 +51,106 @@ def _d(d):
...
+    """Emit a collected datapoint."""
+    val = collectd.Values()
+    val.plugin = 'docker'
+    val.plugin_instance = container['Name']
+
+    # Add additional extracted dimensions through plugin_instance.
+    if dimensions:
+        val.plugin_instance += '[{dims}]'.format(dims=_d(dimensions))
+
+    if type:
```
Comment (mdubbyap) [inline dockerplugin.py]: type is a required field

Comment (AlmightyOatmeal) [inline dockerplugin.py]: a variable shouldn't be named type due to shadowing a built-in function (type()) anyway

**Comment by mdubbyap** [comment]
lgtm

File: dockerplugin.py [hunk 2]
```diff
@@ -51,109 +51,106 @@ def _d(d):
...
-
-class MemoryStats(Stats):
-    @classmethod
-    def read(cls, container, dimensions, stats, t):
-        values = [stats['limit'], stats['max_usage'], stats['usage']]
-        cls.emit(container, dimensions, 'memory.usage', values, t=t)
-
-        for key, value in stats['stats'].items():
-            cls.emit(container, dimensions, 'memory.stats', [value],
+def emit(container, dimensions, type, value, t=None,
```
Comment (AlmightyOatmeal) [inline dockerplugin.py]: variable should be renamed to prevent shadowing the built-in type() function.

**Comment by AlmightyOatmeal** [comment]
Looks good to me.

**Pull request merged**